### PR TITLE
feat: BIP-340 Schnorr Signature 

### PR DESCRIPTION
**Pull request by egeaybars123**
BIP-340 has been implemented in Cairo by the JoyBoy community, and it would be a good addition to add it to Alexandria.

## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/math/src/schnorr.cairo
Comment (maciejka) [inline packages/math/src/schnorr.cairo:1]: Rename the module back to bip-340, it is not a general purpose schnorr, it uses double sha256 as a hash function which is very bitcoin specific.

**Review by maciejka — CHANGES_REQUESTED** [review]

File: packages/math/src/schnorr.cairo
Comment (egeaybars123) [reply packages/math/src/schnorr.cairo]: Done!

**Review by maciejka — APPROVED** [review]

File: packages/math/src/bip340.cairo
Comment (egeaybars123) [reply packages/math/src/bip340.cairo:43]: Thanks! I can commit the changes in a few hours.

Comment (egeaybars123) [reply packages/math/src/bip340.cairo]: @gaetbout Done!

Comment (gaetbout) [reply packages/math/src/bip340.cairo]: Hey, my bad, I should have been more specific...
I believe you can even push it further :)

Comment (egeaybars123) [reply packages/math/src/bip340.cairo]: Ahh I see now hahaha. Will start working on it later today. Thanks!

**Comment by gaetbout** [comment]
@0xLucqs can you also have a look to this?
Would appreciate a second pair of eyes 😄

**Comment by egeaybars123** [comment]
image
This was the best I could come up with @gaetbout :D

**Review by gaetbout — APPROVED** [review]

File: packages/math/src/bip340.cairo
Comment (gaetbout) [inline packages/math/src/bip340.cairo:77]: I don't think that the documentation is following the standard (cfr other files)
Once that is fixed, it'll be merged :)

**Pull request merged**
